chips/sam4l/src/adc.rs

```diff
@@ -332,9 +354,21 @@ impl hil::adc::Adc for Adc {
                     return ReturnCode::FAIL;
                 }
             }
+
+            ReturnCode::SUCCESS
         }
+    }
+}
 
-        ReturnCode::SUCCESS
+/// Implements an ADC capable reading ADC samples on any channel.
+impl hil::adc::Adc for Adc {
+    type Channel = AdcChannel;
+
+    /// Enable and configure the ADC.
+    /// This can be called multiple times with no side effects.
+    fn initialize(&self) -> ReturnCode {
```

Comment:
I can see both sides here, but an additional argument for doing lazy initialization is that, most likely, if we want to support super deep sleep states of various peripherals, we'll anyway need to be able to reinitialize them post-boot.